src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -41,12 +45,14 @@ public class RpcInitializationService : IHostedService
             LanguageWorkerConstants.JavaLanguageWorkerName
         };
 
-        public RpcInitializationService(IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IEnvironment environment, IRpcServer rpcServer, IWebHostLanguageWorkerChannelManager languageWorkerChannelManager, ILogger<RpcInitializationService> logger)
+        public RpcInitializationService(IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IEnvironment environment, IRpcServer rpcServer, IWebHostLanguageWorkerChannelManager languageWorkerChannelManager, IScriptEventManager eventManager, ILogger<RpcInitializationService> logger)
         {
             _applicationHostOptions = applicationHostOptions ?? throw new ArgumentNullException(nameof(applicationHostOptions));
             _logger = logger;
             _rpcServer = rpcServer;
             _environment = environment;
+            _eventManager = eventManager;
+            _eventManager.OfType<ScriptHostStateChangedEvent>().Where(a => a.OldState.Equals(ScriptHostState.Stopping) && a.NewState.Equals(ScriptHostState.Stopped)).Subscribe(KillRpcServer);
             _languageWorkerChannelManager = languageWorkerChannelManager ?? throw new ArgumentNullException(nameof(languageWorkerChannelManager));
```

Comment:
Dispose this subscription in stopasync

Reply:
This subscription needs to live beyond StopAsync() because it's completion is what will lead to StopAsync() of WebJobsScriptHostService which is what will eventually trigger the event that this subscription is listening to. Perhaps if the RpcInitializationService is IDisposable, it might make sense to dispose the same in the Dispose() method...